client/src/components/Filter.jsx

```diff
@@ -10,8 +24,208 @@ import React from "react";
  *
  * Cancel should clear any unapplied filters and close the dropdown
  */
-const Filter = () => {
-  return <></>;
+
+const Filter = ({ filterLabel }) => {
+  const [allFilterData, setAllFilterData] = useContext(FilterContext);
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [icon, setIcon] = useState(<ArrowDropDown />);
+  const [buttonBgColor, setButtonBgColor] = useState("blue.200");
+  const [buttonBorderColor, setButtonBorderColor] = useState("gray.50");
+
+  const [checked, setChecked] = useState(allFilterData[filterLabel].data);
+  const [cancelButton, setCancelButton] = useState("");
+
+  /* buttonStatus
+    0 - all boxes are initially unchecked
+    1 - some or all boxes are initially checked
+    2 - some or all boxes are initially checked, and changes are made
+  */
+  const [buttonStatus, setButtonStatus] = useState(() => {
+    Object.keys(checked).forEach((key) => {
+      if (checked[key].checked) return 1;
+    });
+    return 0;
+  });
+  const handlePopUp = (isOpen, anchor = null) => {
+    if (isOpen) {
+      setAnchorEl(anchor);
+      setIcon(<Close />);
+      setButtonBgColor("red.300");
+      setButtonBorderColor("blue.200");
+    } else {
+      setAnchorEl(anchor);
+      setIcon(<ArrowDropDown />);
+      setButtonBgColor("blue.200");
+      setButtonBorderColor("gray.50");
+    }
+  };
+
+  const handleClick = (event) => {
+    let hasAtLeastOneChecked = false;
+    Object.keys(checked).forEach((key) => {
+      if (checked[key].checked) hasAtLeastOneChecked = true;
+    });
+
+    if (hasAtLeastOneChecked) changeButtonNames(1);
+    else changeButtonNames(0);
+    handlePopUp(true, event.currentTarget);
```

Comment:
maybe just a nit pick but this organization of the inline `if` conditions with an `else` following w/o curlies feels confusing to me. but see other comments about button names first